### PR TITLE
ruby bundle install work around to build packages on mac

### PR DESCRIPTION
**Pull request by apolcyn**
After running `gem update --system` on mac workers, successfully ran the ruby artifact script on them after this change.

### CODE REVIEW

File: grpc.gemspec [hunk 1]
```diff
@@ -27,7 +27,7 @@ Gem::Specification.new do |s|
   s.require_paths = %w( src/ruby/bin src/ruby/lib src/ruby/pb )
   s.platform      = Gem::Platform::RUBY
 
-  s.add_dependency 'google-protobuf', '~> 3.1.0'
+  s.add_dependency 'google-protobuf', '~> 3.1'
```
Comment (nicolasnoble) [inline grpc.gemspec]: Shouldn't that just read "3" instead ?

Comment (murgatroid99) [inline grpc.gemspec]: '~> 3.1' means ">= 3.1 and < 4.0". I don't actually know what would happen if you used '~> 3'.

Comment (nicolasnoble) [inline grpc.gemspec]: Ah, okay. Right. Ruby's semver is weird. Nevermind, thanks :)

**Comment by apolcyn** [comment]
removed the label - was working on the mac issue.

re-running artifact on https://grpc-testing.appspot.com/job/gRPC_build_artifacts/2722/ - mac now passing but linux only failing with issue that should have been fixed on this branch

**Comment by nicolasnoble** [comment]
I just checked your history on your branch, you don't have the fix, no.

**Comment by apolcyn** [comment]
ok, so will merge this so `v1.1.x` should then build

**Comment by apolcyn** [comment]
jenkins: test this please

**Comment by apolcyn** [comment]
jenkins: test this please

**Comment by apolcyn** [comment]
There are now one new ruby failure on mac (still several objc)

The mac ruby failures in `dbg` look to be from using protobuf-3.1 on ruby 2.4 - expected to fail

The ruby mac `opt` failure is odd though
(https://grpc-testing.appspot.com/job/gRPC_pull_requests_macos/2418/testReport/junit/(root)/aggregate_tests/run_tests_ruby_macos_opt/)

**Comment by nicolasnoble** [comment]
*facepalm*

**Comment by nicolasnoble** [comment]
Ruby 2.4.0 is broken, basically.

**Comment by apolcyn** [comment]
jenkins: test this please

**Comment by apolcyn** [comment]
removing -Werror option in extconf.rb worked when building manually.

filed https://bugs.ruby-lang.org/issues/13184 for the ruby.h compile failure

**Comment by apolcyn** [comment]
test failures:
- linux PR: https://github.com/grpc/grpc/issues/9150 cpp_protobuf_async_client_sync_server_streaming_qps_unconstrained_insecure (known issue)
- mac PR: 
  - 2/4 failures from grpc-ruby failing to load `google-protobuf-3.1` on ruby 2.4, but this is expected.
  - other 2 failures noted in new issue: https://github.com/grpc/grpc/issues/9573

**Comment by apolcyn** [comment]
this still ok to merge?

**Pull request merged**